### PR TITLE
Add EVENT_HANDLER_FOREIGN_EVENT lint rule for cross-cluster events

### LINKED ISSUE #824: 3.5.8: Add HANDLER_CREATES_FOREIGN_AGGREGATE lint rule
## Summary

Add a new lint rule `HANDLER_CREATES_FOREIGN_AGGREGATE` (WARNING level) to the 3.5.4 handler completeness rules. This rule detects event handlers that handle events from a different cluster than the one they belong to (`part_of`), which is a structural smell that often indicates a Process Manager would be more appropriate.

## Parent Epic

Sub-issue of #773 (3.5 Architecture Fitness Functions)

## Detection Logic (IR-based, no AST walk)

For each event handler in the IR:
- Check if it handles events from a stream category belonging to Cluster A but is `part_of` Cluster B
- This means the handler is reacting to another aggregate's events — safe when idempotent, but often indicates a Process Manager is a better fit

## Diagnostic Output

```
WARNING [HANDLER_CREATES_FOREIGN_AGGREGATE]: OrderFulfillmentHandler
(part_of: Fulfillment) handles events from stream 'ordering::order'.
Cross-cluster event handling is safe if idempotent, but consider a
ProcessManager if events have causal dependencies.
```

## Details

- **Level**: WARNING (suppressible via `suppress_checks`)
- **Category**: `handler_completeness`
- **Fits into**: 3.5.4 sub-issue scope (handler completeness rules)
- **Rationale**: See ADR-0009 for the full decision record on concurrent event processing strategy

## Depends On

- 3.5.1 (Diagnostic Schema & Suppression Engine) must be implemented first

### PR DESCRIPTION
**Pull request by subhashb**
## Summary
- Adds `EVENT_HANDLER_FOREIGN_EVENT` (warning, category `handler_completeness`) to `protean check`, flagging event handlers whose `part_of` cluster differs from the cluster that owns an event they handle
- Same-cluster handling stays silent, and projectors/process managers are out of scope by design since cross-cluster event consumption is their normal job
- Fixes a false positive where events with colliding `__type__` names across clusters (no single deterministic owner) were incorrectly flagged
- Honours per-element `suppress_checks` and the `[lint.suppressions]` allow-list

## Test plan
- [ ] Core tests pass (`protean test`)
- [ ] `tests/ir/test_elements_contracts_diagnostics.py` covers the new rule, including the colliding-`__type__` edge case

Closes #824

### CODE REVIEW

File: tests/ir/test_elements_contracts_diagnostics.py [hunk 1]
```diff
@@ -2700,3 +2700,422 @@ def test_element_without_module_is_skipped(self):
...
+        domain = Domain(name="FEPm", root_path=".")
+
+        @domain.event(part_of="FlowOrder")
+        class FlowOrderPlaced:
+            order_id = Identifier(required=True)
+
+        @domain.event(part_of="FlowPayment")
+        class FlowPaymentConfirmed:
+            payment_id = Identifier(required=True)
+
```
[the remark this answers is not in the document]
Comment (subhashb) [reply tests/ir/test_elements_contracts_diagnostics.py]: Fixed: added `order_id = Identifier(required=True)` to `FlowPaymentConfirmed` in `test_process_manager_out_of_scope`, matching the pattern used by the `OrderFlow`/`OrderSaga` PM fixtures elsewhere in the file (event declares every field the PM correlates on). Verified the test still passes and still exercises its intended scenario (`EVENT_HANDLER_FOREIGN_EVENT` not raised for a PM spanning two clusters).
